robot-server/robot_server/protocols/protocol_store.py

```diff
@@ -0,0 +1,119 @@
+"""Methods for saving and retrieving protocol files."""
+from dataclasses import dataclass
+from datetime import datetime
+from fastapi import UploadFile
+from logging import getLogger
+from pathlib import Path
+from typing import Dict, List, Sequence
+
+from opentrons.file_runner import ProtocolFileType
+
+log = getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ProtocolStoreEntry:
+    """An entry in the session store, used to construct response models."""
+
+    protocol_id: str
+    protocol_type: ProtocolFileType
+    created_at: datetime
+    files: List[Path]
+
+
+class ProtocolNotFoundError(ValueError):
```

Comment:
I don't think it really matters, but:

```suggestion
class ProtocolNotFoundError(KeyError):
```

or

```suggestion
class ProtocolNotFoundError(LookupError):
```

?